### PR TITLE
Add `emptyBuilder`

### LINKED ISSUE #669: YaruMasterDetailPage: "empty" builder?
It could be a convenient way to handle empty state if the list of pages is asynchronously constructed, for example. Throwing an exception forces conditional building which doesn't look as nice...

```dart
body: pages.isEmpty
  ? YaruCircularProgressIndicator()
  : YaruMasterDetailPage(
      length: pages.length,
      pageBuilder: ...
    ),
```
vs.
```dart
body: YaruMasterDetailPage(
  length: pages.length,
  pageBuilder: ...
  emptyBuilder: (context) => YaruCircularProgressIndicator(),
),
```

See also:
- https://github.com/canonical/firmware-updater/issues/161

### PR DESCRIPTION
**Pull request by d-loose**
Close #669

### CODE REVIEW

File: lib/src/widgets/master_detail/yaru_master_detail_page.dart [hunk 1]
```diff
@@ -163,29 +167,31 @@ class _YaruMasterDetailPageState extends State<YaruMasterDetailPage> {
...
-          previousPaneWidth: _previousPaneWidth,
-          onLeftPaneWidthChange: (width) => _previousPaneWidth = width,
-          appBar: widget.appBar ?? widget.appBarBuilder?.call(context),
-          bottomBar: widget.bottomBar,
-          controller: _controller,
-        ),
-      ),
-    );
+    return widget.length == 0 || widget.controller?.length == 0
+        ? widget.emptyBuilder?.call(context) ?? const SizedBox.shrink()
```
Comment (jpnurmi) [inline lib/src/widgets/master_detail/yaru_master_detail_page.dart:171]: what do you think about moving this inside the scaffold?

i'm thinking about the kind of structure where you have `YaruMasterDetailPage` right below `MaterialApp`. if we have the empty builder inside the `Scaffold`, we wouldn't get errors about missing `Material` etc.

Comment (d-loose) [reply lib/src/widgets/master_detail/yaru_master_detail_page.dart:171]: Good point. How can we handle this in the `YaruNavigationPage`? We have `LayoutBuilder`->`SafeArea`->`Scaffold` there. Is the `SafeArea` needed for anything, and can we just move the `Scaffold` to the top? (On first glance it seems to work, but maybe there are reasons for this widget hierarchy I'm not aware of)

Comment (jpnurmi) [reply lib/src/widgets/master_detail/yaru_master_detail_page.dart]: Not sure where SafeArea came from and even less sure if it does anything on desktop platforms...

**Review by jpnurmi — APPROVED** [review]
LGTM :+1: but let's give @Feichtmeier and @Jupi007 a chance to chime in

**Review by Jupi007 — APPROVED** [review]
Great, LGTM 👍

**Review by Feichtmeier — APPROVED** [review]

**Pull request merged**